### ISSUE TITLE
Reading "Artifact Research Notes" does not fire "ARTIFACT_KNOWLEDGE_CHANGED"

### ISSUE THREAD

**Issue by smaitch** [Rainrider/LibArtifactData-1.0#10]
I would have expected reading the notes would have caused the event to fire.  When opening the artifact UI the event will fire with the new value, but that is suboptimal.

**Comment by Rainrider** [comment]
ARTIFACT_KNOWLEDGE_CHANGED fires when both the knowledge level and multiplier are known. This is not the case when reading the research notes. Register [ARTIFACT_DATA_MISSING](https://github.com/Rainrider/LibArtifactData-1.0/wiki/Events#artifact_data_missing) to get the new knowledge level. For the knowledge multiplier you could use [LAD:ForceUpdate()](https://github.com/Rainrider/LibArtifactData-1.0/wiki/API#forceupdate) or just wait for the user to view an artifact.

**Comment by smaitch** [comment]
Actually I am going to monitor UNIT_SPELLCAST_SUCCEEDED and check for spell ID 219978, which is the reading of the Artifact Research Notes.  Then I just need to check GetCurrencyInfo(1171) which gives me the new artifact knowledge level.

**Comment by Rainrider** [comment]
You can use [CURRENCY_DISPLAY_UPDATE](https://github.com/Rainrider/LibArtifactData-1.0/blob/master/LibArtifactData-1.0.lua#L396-L403), it fires a lot less than UNIT_SPELLCAST_SUCCEEDED. Why you'd opt to do it like that and still use LAD, I don't know. Why do you need that information to be that exact anyway?

**Comment by smaitch** [comment]
I was unaware of CURRENCY_DISPLAY_UPDATE.  I only wanted to use LAD to find out when the artifact knowledge changed.  In testing I found the issue here, so I posted it.  In researching a way around it I found the solution I explained, and have turned off the use of LAD.  I need this information because Grail needs to know when the artifact knowledge has passed a couple of different levels to indicate that some quests in the order hall are now available.
